src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -97,8 +99,94 @@ private void validateTable(TableModel table, Errors errors, SchemaValidationCont
                     }
                 }
             }
+
             context.addTable(tableName, columnNames);
         }
+
+        PartitionStrategy strategy = table.getPartitionStrategy();
+        String partitionColumnName = table.getPartitionColumn();
+        ColumnModel partitionColumn = columns == null ? null : columns.stream()
+            .filter(col -> col.getName().equals(partitionColumnName))
+            .findFirst()
+            .orElse(null);
+        IntPartitionOptionsModel intOptions = table.getIntPartitionOptions();
+
+        if (strategy != null) {
+            if (strategy == PartitionStrategy.INGESTTIME) {
+                if (partitionColumnName != null) {
```

Comment:
As mentioned on the OpenAPI, this would be simpler if you broke out the input into separate objects.